software/nosql/src/main/java/brooklyn/entity/nosql/riak/RiakNodeSshDriver.java

```diff
@@ -112,48 +106,65 @@ public void install() {
-                "which apt-get",
-                ok(sudo("apt-get -y --allow-unauthenticated install logrotate libpam0g-dev libssl0.9.8")),
-                "export OS_NAME=" + Strings.toLowerCase(osDetails.getName()),
-                "export OS_RELEASE=" + osReleaseCmd,
-                String.format("wget -O %s %s", saveAsApt, downloadUrl),
-                sudo(String.format("dpkg -i %s", saveAsApt)));
-        String yum = chainGroup(
-                "which yum",
-                ok(sudo("yum -y install openssl")),
-                String.format("wget -O %s %s", saveAsYum, entity.getAttribute(RiakNode.DOWNLOAD_URL_RHEL_CENTOS)),
-                sudo(String.format("rpm -Uvh %s", saveAsYum)));
+    private List<String> installFromPackageCloud() {
+        OsDetails osDetails = getMachine().getMachineDetails().getOsDetails();
         return ImmutableList.<String>builder()
-                .add("mkdir -p " + installBin)
-                .add(INSTALL_CURL)
-                .add(alternatives(apt, yum))
-                .add("ln -s `which riak` " + Urls.mergePaths(installBin, "riak"))
-                .add("ln -s `which riak-admin` " + Urls.mergePaths(installBin, "riak-admin"))
+                .add(osDetails.getName().toLowerCase().contains("debian") ? addSbinPathCommand() : "")
+                .add(ifNotExecutable("curl", Joiner.on('\n').join(installCurl())))
+                .addAll(ifExecutableElse("yum", installDebianBased(), installRpmBased()))
+                .build();
+    }
+
+    public List<String> installCurl() {
+        return ImmutableList.<String>builder()
+                .add(ifExecutableElse("yum",
+                        BashCommands.sudo("apt-get install --assume-yes curl"),
+                        BashCommands.sudo("yum install -y curl")))
+                .build();
+    }
+
+    private ImmutableList<String> installDebianBased() {
+        return ImmutableList.<String>builder()
+                .add("curl https://packagecloud.io/install/repositories/basho/riak/script.deb | " + BashCommands.sudo("bash"))
+                .add(BashCommands.sudo("apt-get install --assume-yes riak"))
+                .build();
+    }
+
+    private ImmutableList<String> installRpmBased() {
+        return ImmutableList.<String>builder()
+                .add("curl https://packagecloud.io/install/repositories/basho/riak/script.rpm | " + BashCommands.sudo("bash"))
+                .add(BashCommands.sudo("yum install -y riak"))
+                .build();
+    }
+
+    private static String addSbinPathCommand() {
+        return "export PATH=$PATH:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin";
+    }
+
+    /**
+     * Returns a command which
+     * executes <code>statement</code> only if <code>command</code> is NOT found in <code>$PATH</code>
+     *
+     * @param command
+     * @param statement
+     * @return command
+     */
+    private static String ifNotExecutable(String command, String statement) {
```

Comment:
That seems very wrong. As there's nothing special about this entity, does that mean _all_ entities get the wrong BashCommands class? That needs to be fixed asap.